File: book_kit/src/main/java/net/lzbook/kit/ui/widget/ScrollForGridView.java
package net.lzbook.kit.ui.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.GridView;

/**
 * 迁移至book-kit下
 * Created by Administrator on 2017\9\5 0005.
 */
public class ScrollForGridView extends GridView {


        public ScrollForGridView(Context context, AttributeSet attrs) {
            super(context, attrs);
        }

        public ScrollForGridView(Context context) {
            super(context);
        }

        public ScrollForGridView(Context context, AttributeSet attrs, int defStyle) {
            super(context, attrs, defStyle);
        }
        //该自定义控件只是重写了GridView的onMeasure方法，使其不会出现滚动条，ScrollView嵌套ListView也是同样的道理，不再赘述。
        @Override
        public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
            int expandSpec = MeasureSpec.makeMeasureSpec(
                    Integer.MAX_VALUE >> 2, MeasureSpec.AT_MOST);
            super.onMeasure(widthMeasureSpec, expandSpec);
        }

}

File: book_kit/src/main/java/net/lzbook/kit/widget/ScrollForGridView.kt
package net.lzbook.kit.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.GridView


/**
 * Desc：重写GridView的onMeasure方法，使其不会出现滚动条，ScrollView嵌套ListView也是同样的道理
 * Author：JoannChen
 * Mail：yongzuo_chen@dingyuegroup.cn
 * Date：2018/9/20 0020 10:17
 */
class ScrollForGridView : GridView {

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)


    public override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val expandSpec = View.MeasureSpec.makeMeasureSpec(
                Integer.MAX_VALUE shr 2, View.MeasureSpec.AT_MOST)
        super.onMeasure(widthMeasureSpec, expandSpec)
    }

}